chore: update docs based on recent changes to product (#301)
open-sauced/docs: docs/features/star-search.md

- "open source maintainers" 
- "AI"

> ---
> ::: tip
> 
> Currently, StarSearch is in development, but you can [sign up for the waitlist now](https://app.opensauced.pizza/star-search/waitlist).
> 
> :::

Reviewer: This doesn't render well on website as below:

![tip](https://github.com/open-sauced/docs/assets/45172775/127243da-c5fb-45fb-8dd8-40ad32bdf65c)

Let's add a new line after `---` for best practice and remove the space before `tip` to fix the admonition.

```suggestion
---

:::tip

Currently, StarSearch is in development, but you can [sign up for the waitlist now](https://app.opensauced.pizza/star-search/waitlist).

:::
```

## Key Features

- **Real-Time Insights**: Access up-to-date information about GitHub contributions, issues, and projects.
- **Conversational Queries**: Use natural language to explore GitHub data.
- **Cross-Project Analysis**: Identify connections between different projects and technologies.
- **Contributor Discovery**: Pinpoint key contributors and the work they've done

### What Can You Discover with StarSearch?

- **Contributor Activities**: Learn the details of contributions and contributors.
- **Key Contributors**: Identify the key contributors to projects and ecosystems.
- **Work-Based Connections**: Find potential collaborators based on their contributions to specific projects.
- **Hidden Experts**: Discover untapped talent in the ecosystem.

![StarSearch](../../static/img/star-search.png)

## Getting Started with StarSearch

1. Log In to [Your OpenSauced Account](https://app.opensauced.pizza/)

> 2. Navigate to [https://app.opensauced.pizza/star-search](https://app.opensauced.pizza/star-search) or click the StarSearch tab in the sidebar or nav bar.

Author: I'm going to push back a little bit here. I've never heard top bar used before. I think navigation bar would be clearer.

> 2. Navigate to [https://app.opensauced.pizza/star-search](https://app.opensauced.pizza/star-search) or click the StarSearch tab in the sidebar or nav bar.

Reviewer 2: I agree with @BekahHW here.

- **Hidden Experts**: Discover untapped talent in the ecosystem.

![StarSearch](../../static/img/star-search.png)

### Queries Suggestions

> #### **Contributor Activities**
>     - Show me the contributions of \{username}
>     - What \{technology} contributions has \{username} made recently?
>     - What issues has \{username} commented on?
> - **Work-Based Connections**: 
>     - Who has contributed to both project A and project B?
>     - Who are the contributors to both \{technology} and \{technology}?
> - **Hidden Experts**: 
>     - Who are the experts in \{technology}?
>     - Who are the most prevalent contributors to the \{technology} ecosystem?

Reviewer: There are a couple of things here:

- There should be a space after the heading.
- Because there are "Work-Based Connections" and "Hidden Experts", can we group the first three lists in something like "Contribution Activities" or "General Activities"? 

```suggestion
#### **Contributor Activities**

- **Contributions Activities**:
  - Show me the contributions of \{username}
  - What \{technology} contributions has \{username} made recently?
  - What issues has \{username} commented on?
- **Work-Based Connections**:
  - Who has contributed to both project A and project B?
  - Who are the contributors to both \{technology} and \{technology}?
- **Hidden Experts**:
  - Who are the experts in \{technology}?
  - Who are the most prevalent contributors to the \{technology} ecosystem?
```

> #### Technology Insights
>     - What are the most popular projects in the \{technology} ecosystem? 
>     - When was the last release of \{project}?
>     - Whare are the breaking changes in \{project}'s last release?
> 
> ### FAQs
> - **Can you continue conversations with StarSearch?**
>     - Not yet, but stay tuned!
> - **Can you access your search history?**
>     - Not yet.

Reviewer: Let's:

- give a space after the headings here, following Markdown best practice
- fix the indentations
- fix typo

```suggestion
#### Technology Insights

- What are the most popular projects in the \{technology} ecosystem? 
- When was the last release of \{project}?
- Where are the breaking changes in \{project}'s latest release?

### FAQs

- **Can you continue conversations with StarSearch?**
  - Not yet, but stay tuned!
- **Can you access your search history?**
  - Not yet.
```

> - **Can you continue conversations with StarSearch?**
>     - Not yet, but stay tuned!
> - **Can you access your search history?**
>     - Not yet.

Reviewer: **Question**: As FAQ's are usually questions from individuals to us, would it be better to use "I" as a subject here just like our [FAQ's page](https://docs.opensauced.pizza/welcome/faqs/)?

 ```suggestion
- **Can I continue conversations with StarSearch?**
    - Not yet, but stay tuned!
- **Can I access my search history?**
    - Not yet.
```

> 

Reviewer: Let's remove the excess new lines here.
